### DISCUSSION TITLE
Is there a porting guide/documentation?

### DISCUSSION THREAD

**Discussion by EdwinFairchild** [blackmagic-debug/blackmagic#1177] [Documentation]
I would like to use blackmagic debug with  Cortex M based microcontrollers  from Analog devices / Maxim Integrated. I have skimmed the article found [here](https://black-magic.org/hacking/target-drivers.html) and I am wondering if that all it takes? Not that is understood everything but im just asking if I should accomplish everything there would I then have new supported device or is there more to that unfinished article. 

Also would be nice to enable [Github Discussions](https://docs.github.com/en/discussions/quickstart ) for this repo, since this is not really and issue but rather a question open for  discussion. 

Thanks.

**Comment by dragonmux** [comment]
We'll talk with Esden re GH Discussions - that's a fair point.

Regarding the issue at hand, that article is severely outdated now.. it will serve as a decent basis still but you will need to add your new probing routine to target_probe.c and target_probe.h as well and insert a suitable call after the existing routines in cortexm.c (the block in cortexm_probe started at line 415)

For when v1.9 is looking more complete, we will add this to our roster of documentation-related issues that need sorting as it's because of v1.9 that we caused that guide to get outdated. When we do that we'll generally rewrite the guide so it provides a better spring-board for newcommers.

**Comment by esden** [comment]
Here you go, now we have discussions enabled.

As for documentation on adding new targets. This needs work. A great way to write this kind of documentation is actually from the perspective of a newcomer. It is sometimes hard to figure out what is confusing and what needs to be described if one is very familiar with the codebase and it's structure.

So @EdwinFairchild I think you should start working on adding support for your target and ask questions here, and open Q&A discussion topics where appropriate. I hope this will eventually aggregate so that we can collate it into good documentation on the website.

**Reply by EdwinFairchild** [reply]
Thanks for the discussions option, it is just better than flooding the Issues tab  with Q&A. Ill go ahead and start to give this porting a whirl while documenting the process/questions so that perhaps it may pave the way for official documentation.

**Comment by mrx23dot** [comment]
Do chip vendors know about BMP? 
It would be great if they contributed.
